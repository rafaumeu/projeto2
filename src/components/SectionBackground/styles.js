import styled, { css } from 'styled-components'

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colours.primaryColor};
  color: ${theme.colours.white};
`
export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colours.white};
    color: ${theme.colours.primaryColor};
    ${background && containerBackgroundActivate(theme)}
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`
