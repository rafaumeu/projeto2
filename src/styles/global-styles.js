import { createGlobalStyle } from 'styled-components'

import { theme } from './theme'

export const GlobalStyles = createGlobalStyle`

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  font-size: 62.5%;
}
body {
  font-size: 1.6rem;
  font-family: ${({ theme }) => theme.fonts.family.primary}
}
h1, h2, h3, h4, h5, h6 {
  font-family: ${({ theme }) => theme.fonts.family.secondary};

  margin: ${({ theme }) => theme.spacings.sizes.large} 0;
}
p {
  margin: ${({ theme }) => theme.spacings.sizes.medium} 0;
}
ul, ol {
  margin: ${({ theme }) => theme.spacings.sizes.medium} ;
  padding: ${({ theme }) => theme.spacings.sizes.medium} ;
}
a {
  color: ${({ theme }) => theme.colours.secondaryColor}
}
`
