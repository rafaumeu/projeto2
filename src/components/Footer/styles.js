import styled, { css } from 'styled-components'

import { Container as SectionContainer } from '../SectionContainer/styles'
import { Container as TextComponent } from '../TextComponent/styles'
export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colours.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`
