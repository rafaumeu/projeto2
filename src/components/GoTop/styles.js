import styled, { css } from 'styled-components'

import { theme } from '../../styles/theme'

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    background: ${theme.colours.primaryColor};
    color: ${theme.colours.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    opacity: 0.7;
  `}
`
