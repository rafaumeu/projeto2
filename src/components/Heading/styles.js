import styled, { css } from 'styled-components'

import { theme } from '../../styles/theme'

export const Title = styled.h1`
  ${({ theme, light }) => css`
    color: ${light ? theme.colours.white : theme.colours.primaryColor};
  `}
`
