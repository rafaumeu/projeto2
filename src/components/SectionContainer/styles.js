import styled, { css } from 'styled-components'

import { theme } from '../../styles/theme'

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    margin: 0 auto;
    padding: ${theme.spacings.sizes.large};
  `}
`
