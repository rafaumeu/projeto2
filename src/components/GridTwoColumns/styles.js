import styled, { css } from 'styled-components'

import { Title } from '../Heading/styles'
export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr, 2fr;
    align-items: center;
    gap: ${theme.spacings.sizes.large};
    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.sizes.xlarge};
    }
  `}
`
export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.sizes.large};
  `}
`
export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`
export const Image = styled.img`
  ${({ theme }) => css``}
`
