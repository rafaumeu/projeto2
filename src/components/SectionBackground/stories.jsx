import SectionBackground from '.'
import SectionContainer from '../SectionContainer'

export default {
  title: 'SectionBackground',
  component: SectionBackground,
  args: {
    children: (
      <SectionContainer>
        <div>
          <h1>SectionContainer</h1>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Sed quod
            vitae mollitia animi quidem. Exercitationem in qui commodi minima
            dolores maiores earum, cum, sit architecto iure tempore nostrum
            velit sapiente.
          </p>
        </div>
      </SectionContainer>
    ),
  },
  argTypes: {
    children: { type: '' },
  },
}

export const Template = (args) => {
  return (
    <div>
      <SectionBackground {...args} />
    </div>
  )
}
