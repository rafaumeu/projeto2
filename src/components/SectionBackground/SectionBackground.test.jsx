import { screen } from '@testing-library/react'

import SectionBackground from '.'
import { renderTheme } from '../../styles/render-theme'

describe('<SectionBackground />', () => {
  it('should render with background dark', () => {
    const { container } = renderTheme(
      <SectionBackground background={true}>Children</SectionBackground>
    )
    expect(screen.getByText(/children/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
  it('should render with background light', () => {
    const { container } = renderTheme(
      <SectionBackground>Children</SectionBackground>
    )
    expect(screen.getByText(/children/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
