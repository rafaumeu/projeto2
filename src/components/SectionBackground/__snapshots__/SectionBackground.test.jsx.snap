// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render with background dark 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #ffffff;
  color: #0a1128;
  background: #0a1128;
  color: #ffffff;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Children
    </div>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with background light 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #ffffff;
  color: #0a1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Children
    </div>
  </div>
</div>
`;
